graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/tracing/FederatedTracingInstrumentation.java

```diff
@@ -0,0 +1,282 @@
+package com.apollographql.federation.graphqljava.tracing;
+
+import com.google.protobuf.Timestamp;
+import graphql.ExecutionResult;
+import graphql.ExecutionResultImpl;
+import graphql.GraphQLError;
+import graphql.GraphqlErrorBuilder;
+import graphql.execution.DataFetcherResult;
+import graphql.execution.ExecutionPath;
+import graphql.execution.ExecutionStepInfo;
+import graphql.execution.instrumentation.InstrumentationContext;
+import graphql.execution.instrumentation.InstrumentationState;
+import graphql.execution.instrumentation.SimpleInstrumentation;
+import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters;
+import graphql.execution.instrumentation.parameters.InstrumentationFieldFetchParameters;
+import graphql.schema.DataFetchingEnvironment;
+import mdg.engine.proto.Reports;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static graphql.execution.instrumentation.SimpleInstrumentationContext.whenCompleted;
+import static graphql.schema.GraphQLTypeUtil.simplePrint;
+
+public class FederatedTracingInstrumentation extends SimpleInstrumentation {
+    private static final String KEY = "ftv1";
+    private static final String DEBUG_KEY = "ftv1_debug";
+
+    private boolean debuggingEnabled = false;
+
+    public FederatedTracingInstrumentation() {
+    }
+
+    public FederatedTracingInstrumentation(boolean debug) {
+        debuggingEnabled = debug;
+    }
+
+    @Override
+    public InstrumentationState createState() {
+        return new FederatedTracingState();
+    }
+
+    @Override
+    public CompletableFuture<ExecutionResult> instrumentExecutionResult(ExecutionResult executionResult, InstrumentationExecutionParameters parameters) {
+        FederatedTracingState state = parameters.getInstrumentationState();
+
+        Map<Object, Object> extensions = executionResult.getExtensions();
+        Map<Object, Object> tracingMap = new LinkedHashMap<>(extensions == null ? Collections.emptyMap() : extensions);
+
+        Reports.Trace trace = state.toProto();
+
+        byte[] bytes = trace.toByteArray();
+        tracingMap.put(KEY, Base64.getEncoder().encodeToString(bytes));
+
+        if (debuggingEnabled) {
```

Comment:
I like this feature, though I think it might be more useful as a log than as a separate extensions key 🤷‍♂ curious on your thoughts!